documentation/client-side-forms/tutorial-validation-ts.asciidoc

```diff
@@ -16,13 +16,88 @@ For each model, there is an associated `BinderNode`, which you can add validator
 BinderNode name =  binder.for(this.binder.model.name);
 ----
 
-The `binder` itself is also the root level `BinderNode`. So Validators can be added any level of the data structure; hence it's possible to have both field-level and 
+The `binder` itself is also the root level `BinderNode`. So Validators can be added to any level of the data structure; hence it's possible to have both field-level and 
 record-level validators.
 
 === Field-level Validator
 
 A field-level validator is a validator added to a child `BinderNode`, which binds to a UI field. It will be executed either when the field value changes or when the whole form is validated before submitting.
 
+==== Built-in validators
```

Comment:
Most of the descriptions below were mostly copied from the linked JSR 380 spec.

However I noticed that the behaviour of some of the validators here seem to differ a bit from what is described in the spec even though the source code also has [a comment](https://github.com/vaadin/flow/blob/master/flow-client/src/main/resources/META-INF/resources/frontend/form/Validators.ts#L43) saying they should be equivalent. Notably Javadoc for most of the constraints (except `NotNull`, `NotEmpty`, `NotBlank`) says that `null` is considered a valid value, but I don't think it's the case in some of the client side validators (though I didn't test).

I think the list of validators and their supported options should be documented a bit more verbosely, but then this list should probably be extracted to a separate file. That can be done as a future enhancement.